projects/js-packages/publicize-components/src/components/schedule-button/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { Dropdown, Button, DateTimePicker } from '@wordpress/components';
+import { date, getSettings } from '@wordpress/date';
+import { useCallback, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { calendar } from '@wordpress/icons';
+import styles from './styles.module.scss';
+
+interface ScheduleButtonBaseProps {
+	scheduleTimestamp?: number;
+	onChange?: ( unixTimestamp: number ) => void;
+	onConfirm?: () => void;
+}
+
+interface ScheduleButtonContentProps extends ScheduleButtonBaseProps {
+	onClose: () => void;
+}
+
+interface ScheduleButtonProps extends ScheduleButtonBaseProps {
+	isBusy?: boolean; // Defaults to false
+}
+
+const ScheduleButtonContent = ( {
+	onClose,
+	scheduleTimestamp,
+	onChange,
+	onConfirm,
+}: ScheduleButtonContentProps ) => {
+	const confirmCalback = useCallback( () => {
+		onConfirm?.();
+		onClose();
+	}, [ onClose, onConfirm ] );
+
+	const changeCallback = useCallback(
+		newDate => {
+			const unixTime = +date( 'U', newDate + getSettings().timezone.abbr, 0 );
+			onChange?.( unixTime );
+		},
+		[ onChange ]
+	);
+
+	const scheduleDate = useMemo(
+		() =>
+			date(
+				'Y-m-d\\TH:i:s',
+				scheduleTimestamp ? new Date( scheduleTimestamp * 1000 ) : new Date(),
+				getSettings().timezone.offset
+			),
+		[ scheduleTimestamp ]
+	);
```

Comment:
This computation shouldn't be that expensive to warrant `useMemo`, but it's fine I think.

Reply:
Agreed. I think that might have been a hangover from something, or me just over-optimising!